### PR TITLE
Personal combat messages

### PR DESCRIPTION
**Pull request by kingofkosmos**
## About The Pull Request
Most of the time in a combat situation there's 3 sides involved: the attacker, the victim and the observer.

Right now either the attacker or the victim always have to see an "observer-message", showing them their own name, which is kinda silly, why not just refer to them as "you"? You shouldn't have to be told your name over and over again in the game.

So this change adds the attacker to the ignored list of the observer-message and adds a personal message for them. It only involves the basic unarmed attacks so if this doesn't work, it's not too bad to revert. Although I tested it and it seemed to work fine.

How people see messages involving combat right now:
Untitled

How people would see messages involving combat after:
Untitled

If this proves to work without any problems, I'll make all the other combat messages more personal as well.

## Why It's Good For The Game
Less chance of misreading what's happening in the combat, who's getting hit by who and such. Makes the game feel more polished.

### CODE REVIEW

**Comment by Qustinnus** [comment]
There's a second arg to visible message that sends a message to the source, aka the attacker.

**Comment by Qustinnus** [comment]
Oh wait I forgot attack code sucks and half of it does not use the attacker as source but the defender lmao

**Comment by Qustinnus** [comment]
Maybe go through the martial arts datums too?

**Comment by kingofkosmos** [comment]
> There's a second arg to visible message that sends a message to the source, aka the attacker.
> Oh wait I forgot attack code sucks and half of it does not use the attacker as source but the defender lmao

Yes, but that only covers the attacker/victim + observer(s). Either the attacker or victim are going to get the observer-message, which is what I'm trying to go around with this. So all attacker + victim + observer(s) would be covered with their own messages.

**Comment by zxaber** [comment]
I long for the day that I can put my own name in the highlighter without getting all the attack message false positives.

**Comment by nemvar** [comment]
I don't like how all of this is present tense. Feels like everything in the chat should be simple past.

**Comment by Qustinnus** [comment]
small idea: maybe make a proc on mob that's called attack_message that takes an attacker, attacked, attacker message and attacked message. or something like that, allows for a more generic approach and cant be fucked up in the future

**Comment by kingofkosmos** [comment]
That's an excellent idea, but first I'd rather make a working feature, and then refine/rework it into a new proc. If that's ok. Getting too fancy too quick, with too little experience, lead me previously to a lot of alt-clicking issues with machinery.

**Comment by kingofkosmos** [comment]
This is ready for reviewing.

**Comment by optimumtact** [comment]
no it's not

**Comment by kingofkosmos** [comment]
I somehow didn't see those conflicts yesterday. They should be ok now!

File: code/modules/mob/living/carbon/alien/alien_defense.dm [hunk 1]
```diff
@@ -38,7 +38,8 @@ In all, this is a lot like the monkey code. /N
 				M.do_attack_animation(src, ATTACK_EFFECT_BITE)
 				playsound(loc, 'sound/weapons/bite.ogg', 50, TRUE, -1)
 				visible_message("<span class='danger'>[M.name] bites [src]!</span>", \
-						"<span class='userdanger'>[M.name] bites you!</span>", null, COMBAT_MESSAGE_RANGE)
+								"<span class='userdanger'>[M.name] bites you!</span>", null, COMBAT_MESSAGE_RANGE, M)
```
Comment (kingofkosmos) [inline code/modules/mob/living/carbon/alien/alien_defense.dm]: Btw sometimes there's [M.name] used in the messages, but usually it's just [M].
Should [M.name] be changed to [M], or vice versa, or does it matter?

**Comment by Fhaxaris** [comment]
>agent id forge to "You"

what happens?

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> > agent id forge to "You"
> 
> what happens?

You punch You

File: code/modules/mob/living/carbon/alien/alien_defense.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/alien/alien_defense.dm]: M and M.name should be the same, it doesn't pull real_name or anything like that.

**Comment by ExcessiveUseOfCobblestone** [comment]
how does it prevent the attacker from seeing the visible message or am I missing something?

It seems like having 2 of the same message would be more confusing that your name vs. you

**Comment by kingofkosmos** [comment]
> how does it prevent the attacker from seeing the "observer" message or am I missing something?

The last argument in visible_message-proc is "list/ignored_mobs", which makes it so the attacker doesn't receive anything from the visible_message, and only receives the to_chat-message after.

**Comment by AnturK** [comment]
Idea is fine but not sure if this shouldn't have some wrapper so it's more obvious why these two lines exist.

**Comment by optimumtact** [comment]
visible message already has a self parameter, why is that not used?

it goes to the source of the message.

this to me indicates the messages are being generated by the wrong mob

**Comment by kingofkosmos** [comment]
>visible message already has a self parameter, why is that not used?

What do you mean? It is being used. But what only checks two perspectives on a fight, the viewer and the victim. This adds a third message for the attacker.

I know there are better ways to do this, but they are all a bit too much of a rework. This is simple and it seems to work. If you think it’s confusing, should I add a comment after the to_chats explaining them?

**Comment by Qustinnus** [comment]
you should probably just make a new proc that sends combat messages and handles all 3 messages

**Comment by optimumtact** [comment]
all four messages, dont' forget blind people

**Comment by Qustinnus** [comment]
you sense that (x) was punched

**Comment by optimumtact** [comment]
you hear the sickening sound of flesh hitting flesh

**Comment by kingofkosmos** [comment]
I'll add the audible-messages (the third argument), hopefully that would be enough for this PR? I'm not sure if I have the time or the competence to work these into one proc.

File: code/modules/mob/living/carbon/alien/alien_defense.dm (hunk 1 above)
Comment (kingofkosmos) [reply code/modules/mob/living/carbon/alien/alien_defense.dm]: Is there any reason for this to be changed to just [M]? And if so, should all [M]'s in visible_messages be changed to [M.name]? Which one would be better practice in visible_messages?

Comment (Qustinnus) [reply code/modules/mob/living/carbon/alien/alien_defense.dm]: [M]

**Comment by optimumtact** [comment]
I was talking about the arguments to a theorised attack_message proc, you really didn't have to do all that extra work man

**Comment by ExcessiveUseOfCobblestone** [comment]
this man is sick

**Pull request merged**